streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -43,6 +47,9 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
   private val ssc = jobScheduler.ssc
   private val conf = ssc.conf
   private val graph = ssc.graph
+  private lazy val completedBothCheckpoints = new HashMap[Time, Boolean]()
```

Comment:
Ah, makes sense. Do you want me to pull in your commit into this one? That makes sense.